apps/src/templates/Dialog.jsx

```diff
@@ -0,0 +1,251 @@
+import React from 'react';
+import Radium from 'radium';
+import Button from './Button';
+import BaseDialog from './BaseDialog';
+import color from '../color';
+import {childrenOfType, whenNoChildOfTypes} from '../propTypes';
+import locale from '../locale';
+
+const styles = {
+  icon: {
+    backgroundSize: 100,
+    backgroundPosition: 'bottom',
+    backgroundRepeat: 'no-repeat',
+    marginTop: -165,
+    width: 120,
+    height: 200,
+    float: 'left',
+  },
+  title: {
+    fontSize: 27,
+    lineHeight: '30px',
+    marginBottom: 15,
+    color: color.purple,
+  },
+  body: {
+    fontSize: 24,
+    lineHeight: '27px',
+    color: color.purple,
+  },
+  buttons: {
+    marginTop: 10,
+  },
+  rightButton: {
+    float: 'right',
+  },
+  buttonClear: {
+    clear: 'both'
+  },
+  footer: {
+    borderColor: color.light_gray,
+    borderStyle: 'solid',
+    borderWidth: 0,
+    borderTopWidth: 2,
+    marginTop: 15,
+    paddingTop: 15,
+  },
+};
+
+export function Title({children}) {
+  return (
+    <h1 style={styles.title}>
+      {children}
+    </h1>
+  );
+}
+
+export function Body({children}) {
+  return (
+    <div style={styles.body}>
+      {children}
+    </div>
+  );
+}
+
+export function Confirm(props) {
+  return <Button type="primary" {...props}>{props.children || locale.dialogOK()}</Button>;
+}
+
+export function Cancel(props) {
+  return <Button type="cancel" {...props}>{props.children || locale.dialogCancel()}</Button>;
+}
+
+export function Buttons({children}) {
+  children = React.Children.toArray(children);
+  var leftChildren = children.slice(0, children.length-1);
+  var rightChild = children[children.length - 1];
+  return (
+    <div style={styles.buttons}>
+      <div style={styles.rightButton}>
+        {rightChild}
+      </div>
+      {leftChildren}
+      <div style={styles.buttonClear}/>
+    </div>
+  );
+}
+Buttons.propTypes = {
+  children: childrenOfType(Cancel, Confirm),
+};
+
+export function Footer({children}) {
+  return (
+    <div style={styles.footer}>
+      {children}
+    </div>
+  );
+}
+
+
+export var Icon = Radium(function Icon({src}) {
+  return (
+    <span style={[styles.icon, {backgroundImage: `url(${src})`}]}/>
+  );
+});
+Icon.propTypes = {
+  src: React.PropTypes.string,
+};
+
+var Dialog = React.createClass({
+  propTypes: Object.assign({}, BaseDialog.propTypes, {
+    children: childrenOfType(
+      Icon,
+      Title,
+      Body,
+      Buttons,
+      Footer
```

Comment:
Okay, so enumerating valid configurations...
```
* [none]
* Icon
* Title
  Body
* Buttons
* Footer
* Icon, Title
  Icon, Body
* Icon, Buttons
* Icon, Footer
  Title, Body
? Title, Buttons
* Title, Footer
  Body, Buttons
  Body, Footer
* Buttons, Footer
  Icon, Title, Body
...etc
```
The starred configurations seem bad to me.  `Title, Buttons` is a little bit of an edge case, but I think we should enforce `Body, Buttons` as the better version of that case (as in your screenshots)... so actually, I think `Body` should be a required child, and it's not with the current validator.